spyder/plugins/updatemanager/workers.py

```diff
@@ -229,15 +229,18 @@ class WorkerDownloadInstaller(BaseWorker):
         Total size of the file expected to be downloaded.
     """
 
-    def __init__(self, latest_release, installer_path, installer_size_path):
+    def __init__(self, latest_release, installer_paths):
         super().__init__()
         self.latest_release = latest_release
-        self.installer_path = installer_path
-        self.installer_size_path = installer_size_path
+        self.installer_paths = installer_paths
         self.error = None
         self.cancelled = False
         self.paused = False
 
+        self.installer_dir = osp.dirname(
+            next(iter(installer_paths.values()))['file']
+        )
```

Comment:
I don't understand what this code does. Could you rewrite it in a more standard way? Or at least could you write a comment about it to explain it? Thanks!